src/worker.c

```diff
@@ -103,6 +123,31 @@ static int init_worker(struct worker_data *data)
 	}
 
 
+        /* Enable periodic worker reporting if enabled on CLI */
+        if (data->worker_report_interval > 0) {
+                fprintf(stderr, "Enable reporting on write pipe fd: %d\n", data->pipe_w); 
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_PROGRESSFUNCTION, old_prog_report)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_PROGRESSDATA, data)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+
+#if LIBCURL_VERSION_NUM >= 0x72000
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_XFERINFOFUNCTION, prog_report)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_XFERINFODATA, data)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+#endif
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_NOPROGRESS, 0L)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+                data->last_dlnow = (curl_off_t)0;
```

Comment:
Heh, think the compiler is smart enough to apply the right type to constants? :)